odxtools/encoding.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: MIT
+from enum import Enum
+from typing import Optional
+
+from .exceptions import odxraise
+from .odxtypes import DataType
+
+
+class Encoding(Enum):
+    BCD_P = "BCD-P"
+    BCD_UP = "BCD-UP"
+
+    ONEC = "1C"
+    TWOC = "2C"
+    SM = "SM"
+
+    UTF8 = "UTF-8"
+    UCS2 = "UCS-2"
+    ISO_8859_1 = "ISO-8859-1"
+    ISO_8859_2 = "ISO-8859-2"
+    WINDOWS_1252 = "WINDOWS-1252"
+
+    NONE = "NONE"
+
+
+def get_string_encoding(base_data_type: DataType, base_type_encoding: Optional[Encoding],
```

Comment:
Should this be moved to DiagCodedType ?

Reply:
unfortuntely, this is not possible because the `DiagCodedType` object is not available in `EncodeState` and `DecodeState`